src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/MLDsa/MLDsaTestHelpers.cs

```diff
@@ -465,6 +465,16 @@ internal static byte[] DoTryUntilDone(TryExportFunc func)
             return buffer.AsSpan(0, written).ToArray();
         }
 
+        // CryptographicException can only have both HRESULT and Message set starting in .NET Core 3.0+.
+        // To work around this, the product code throws an exception derived from CryptographicException
+        // that has both set. This assert checks for that instead.
+        internal static void AssertThrowsCryptographicExceptionWithHResult(Action export)
+        {
+            CryptographicException ce = Assert.ThrowsAny<CryptographicException>(export);
+            Assert.Equal(NTE_NOT_SUPPORTED, ce.HResult);
+            Assert.Contains("The requested operation is not supported.", ce.Message);
```

Comment:
This string comes from the OS, which means it depends on what the user locale/language is... and it's subject to Windows servicing, not us.

You can add a "do I think English is the right answer?" guard (there are some sprinkled throughout the test code)... but I'd just delete the string matching and go with HRESULT validation only.